httpe/err_test.go

```diff
@@ -0,0 +1,43 @@
+package httpe
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestWriteHTTPeErr(t *testing.T) {
+	errs := []error{
+		ErrBadRequest, ErrUnauthorized,
+		ErrPaymentRequired, ErrForbidden, ErrNotFound,
+		ErrMethodNotAllowed, ErrNotAcceptable,
+		ErrProxyAuthRequired, ErrRequestTimeout,
+		ErrConflict, ErrGone, ErrLengthRequired,
+		ErrPreconditionFailed, ErrRequestEntityTooLarge,
+		ErrRequestURITooLong, ErrUnsupportedMediaType,
+		ErrRequestedRangeNotSatisfiable, ErrExpectationFailed,
+		ErrTeapot, ErrMisdirectedRequest,
+		ErrUnprocessableEntity, ErrLocked,
+		ErrFailedDependency, ErrTooEarly,
+		ErrUpgradeRequired, ErrPreconditionRequired,
+		ErrTooManyRequests, ErrRequestHeaderFieldsTooLarge,
+		ErrUnavailableForLegalReasons, ErrInternalServerError,
+		ErrNotImplemented, ErrBadGateway,
+		ErrServiceUnavailable, ErrGatewayTimeout,
+		ErrHTTPVersionNotSupported, ErrVariantAlsoNegotiates,
+		ErrInsufficientStorage, ErrLoopDetected,
+		ErrNotExtended, ErrNetworkAuthenticationRequired,
+	}
+	for _, err := range errs {
+		w := httptest.NewRecorder()
+		WriteHTTPeErr(w, err)
+		require.Contains(t, err.Error(), http.StatusText(w.Code))
```

Comment:
I think you also need to test that an error wrapping an internal error does not get that internal error string written.

Reply:
agreed.

Reply:
done.

Reply:
Maybe I'm missing something, but I don't see that anywhere.

Reply:
oh ok. sorry I didn't read properly.
i just read the wrapping part and thought i got it.... i thought you wanted me to test that the (un-)wrapping works.
sorry. fixing now.

Reply:
done.